File: hackathon_CPP_SDK/src-gen/nevonex/platform_service/Platform_Service.cpp
/*
 * nevonex/platform_service/Platform_Service.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "Platform_Service.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/platform_service/Platform_servicePackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

/*PROTECTED REGION ID(Platform_Service.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::platform_service;
using namespace ::nevonex::log;

// Default constructor
Platform_Service::Platform_Service() : m_download(0), m_uploadData(0), m_uploadFile(0), m_receive(0), m_sendCommand(0), m_sendFile(0)
{
    // Machine does NOT call _initialize() — Provider manages initialization
    m_interfaceDetailMap = {
    };

    /*PROTECTED REGION ID(Platform_Service__Platform_Service) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/
}

Platform_Service::~Platform_Service()
{
    /*PROTECTED REGION ID(Platform_Service__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/
}

// References
::ecore::EString Platform_Service::getDownload() const
{
    return m_download;
}

void Platform_Service::setDownload(::ecore::EString _download)
{
    ::ecore::EString _old_download = m_download;
    m_download = _download;
    SDK_FCAL_LOG(SeverityLevel::debug) << "Platform_Service property \"download\" has changed.";
    notifyPropertyChange("download", _old_download, m_download);
}

::ecore::EString Platform_Service::getUploadData() const
{
    return m_uploadData;
}

void Platform_Service::setUploadData(::ecore::EString _uploadData)
{
    ::ecore::EString _old_uploadData = m_uploadData;
    m_uploadData = _uploadData;
    SDK_FCAL_LOG(SeverityLevel::debug) << "Platform_Service property \"uploadData\" has changed.";
    notifyPropertyChange("uploadData", _old_uploadData, m_uploadData);
}

::ecore::EString Platform_Service::getUploadFile() const
{
    return m_uploadFile;
}

void Platform_Service::setUploadFile(::ecore::EString _uploadFile)
{
    ::ecore::EString _old_uploadFile = m_uploadFile;
    m_uploadFile = _uploadFile;
    SDK_FCAL_LOG(SeverityLevel::debug) << "Platform_Service property \"uploadFile\" has changed.";
    notifyPropertyChange("uploadFile", _old_uploadFile, m_uploadFile);
}

::ecore::EString Platform_Service::getReceive() const
{
    return m_receive;
}

void Platform_Service::setReceive(::ecore::EString _receive)
{
    ::ecore::EString _old_receive = m_receive;
    m_receive = _receive;
    SDK_FCAL_LOG(SeverityLevel::debug) << "Platform_Service property \"receive\" has changed.";
    notifyPropertyChange("receive", _old_receive, m_receive);
}

::ecore::EString Platform_Service::getSendCommand() const
{
    return m_sendCommand;
}

void Platform_Service::setSendCommand(::ecore::EString _sendCommand)
{
    ::ecore::EString _old_sendCommand = m_sendCommand;
    m_sendCommand = _sendCommand;
    SDK_FCAL_LOG(SeverityLevel::debug) << "Platform_Service property \"sendCommand\" has changed.";
    notifyPropertyChange("sendCommand", _old_sendCommand, m_sendCommand);
}

::ecore::EString Platform_Service::getSendFile() const
{
    return m_sendFile;
}

void Platform_Service::setSendFile(::ecore::EString _sendFile)
{
    ::ecore::EString _old_sendFile = m_sendFile;
    m_sendFile = _sendFile;
    SDK_FCAL_LOG(SeverityLevel::debug) << "Platform_Service property \"sendFile\" has changed.";
    notifyPropertyChange("sendFile", _old_sendFile, m_sendFile);
}

// FCAL timestamp/valid
::ecore::ELong Platform_Service::getDownload_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean Platform_Service::isDownload_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong Platform_Service::getUploadData_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean Platform_Service::isUploadData_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong Platform_Service::getUploadFile_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean Platform_Service::isUploadFile_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong Platform_Service::getReceive_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean Platform_Service::isReceive_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong Platform_Service::getSendCommand_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean Platform_Service::isSendCommand_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong Platform_Service::getSendFile_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean Platform_Service::isSendFile_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

// PropertyChange delegation (diamond inheritance requires explicit override)
void Platform_Service::addPropertyChangeListener(
        ::nevonex::types::PropertyChangeListener _listener)
{
    using namespace ::nevonex::types;
    PropertyChange::addPropertyChangeListener(_listener);
}

void Platform_Service::removePropertyChangeListener(
        ::nevonex::types::PropertyChangeListener _listener)
{
    using namespace ::nevonex::types;
    PropertyChange::removePropertyChangeListener(_listener);
}

void Platform_Service::notifyPropertyChange(::ecore::EString const &_name,
        ::ecore::EJavaObject const &_oldValue,
        ::ecore::EJavaObject const &_newValue)
{
    using namespace ::nevonex::types;
    PropertyChange::notifyPropertyChange(_name, _oldValue, _newValue);
}

File: com.bosch.fsp.hackathon.gen/src/com/bosch/nevonex/platform_service/impl/Platform_Service.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.platform_service.impl;

import com.bosch.fsp.logger.FCALLogs;
import com.bosch.fsp.logger.LoggerConstants;

import com.bosch.fsp.platform.InterfaceDetails;

import com.bosch.fsp.runtime.feature.IMachine;

import com.bosch.nevonex.common.ITopicObject;
import com.bosch.nevonex.common.impl.TopicObject;

import com.bosch.nevonex.platform_service.IPlatform_Service;

import com.bosch.nevonex.types.IPropertyChange;
import com.bosch.nevonex.types.impl.TypesPackage;

import java.beans.PropertyChangeEvent;
import java.beans.PropertyChangeListener;

import java.lang.reflect.InvocationTargetException;

import java.util.Collection;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.eclipse.emf.common.util.BasicEList;
import org.eclipse.emf.common.util.EList;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.util.EDataTypeUniqueEList;

/**
 * An implementation of the model object '<em><b>Platform_Service</b></em>'.
 * @generated
 */
public class Platform_Service extends TopicObject implements IPlatform_Service {
	protected EList<PropertyChangeListener> listeners;
	protected Map<String, InterfaceDetails> interfaceDetailMap = new HashMap<>();
	protected String download = null;
	protected String uploadData = null;
	protected String uploadFile = null;
	protected String receive = null;
	protected String sendCommand = null;
	protected String sendFile = null;

	/**
	 * @generated
	 */
	protected Platform_Service() {
		super();
		initInterfaceDetails();
	}

	private void initInterfaceDetails() {
		// Interface details initialization
	}

	/**
	 * @generated
	 */
	@Override
	protected EClass eStaticClass() {
		return Platform_servicePackage.Literals.PLATFORM_SERVICE;
	}

	public String getDownload() {
		return download;
	}

	public void setDownload(String newDownload) {
		download = newDownload;
	}

	public String getUploadData() {
		return uploadData;
	}

	public void setUploadData(String newUploadData) {
		uploadData = newUploadData;
	}

	public String getUploadFile() {
		return uploadFile;
	}

	public void setUploadFile(String newUploadFile) {
		uploadFile = newUploadFile;
	}

	public String getReceive() {
		return receive;
	}

	public void setReceive(String newReceive) {
		receive = newReceive;
	}

	public String getSendCommand() {
		return sendCommand;
	}

	public void setSendCommand(String newSendCommand) {
		sendCommand = newSendCommand;
	}

	public String getSendFile() {
		return sendFile;
	}

	public void setSendFile(String newSendFile) {
		sendFile = newSendFile;
	}

	/**
	 * @generated
	 */
	public void addPropertyChangeListener(PropertyChangeListener listener) {
		if (listeners == null) {
			listeners = new BasicEList<>();
		}
		listeners.add(listener);
	}

	/**
	 * @generated
	 */
	public void removePropertyChangeListener(PropertyChangeListener listener) {
		if (listeners != null) {
			listeners.remove(listener);
		}
	}

	/**
	 * @generated
	 */
	public void notifyPropertyChange(String name, Object oldValue, Object newValue) {
		if (listeners != null) {
			for (PropertyChangeListener listener : this.listeners) {
				listener.propertyChange(new PropertyChangeEvent(this, name, oldValue, newValue));
			}
		}
	}

	/**
	 * @generated
	 */
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case Platform_servicePackage.I_PLATFORM_SERVICE__DOWNLOAD:
			return getDownload();
		case Platform_servicePackage.I_PLATFORM_SERVICE__UPLOAD_DATA:
			return getUploadData();
		case Platform_servicePackage.I_PLATFORM_SERVICE__UPLOAD_FILE:
			return getUploadFile();
		case Platform_servicePackage.I_PLATFORM_SERVICE__RECEIVE:
			return getReceive();
		case Platform_servicePackage.I_PLATFORM_SERVICE__SEND_COMMAND:
			return getSendCommand();
		case Platform_servicePackage.I_PLATFORM_SERVICE__SEND_FILE:
			return getSendFile();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	/**
	 * @generated
	 */
	@SuppressWarnings("unchecked")
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case Platform_servicePackage.I_PLATFORM_SERVICE__DOWNLOAD:
			setDownload((String) newValue);
			return;
		case Platform_servicePackage.I_PLATFORM_SERVICE__UPLOAD_DATA:
			setUploadData((String) newValue);
			return;
		case Platform_servicePackage.I_PLATFORM_SERVICE__UPLOAD_FILE:
			setUploadFile((String) newValue);
			return;
		case Platform_servicePackage.I_PLATFORM_SERVICE__RECEIVE:
			setReceive((String) newValue);
			return;
		case Platform_servicePackage.I_PLATFORM_SERVICE__SEND_COMMAND:
			setSendCommand((String) newValue);
			return;
		case Platform_servicePackage.I_PLATFORM_SERVICE__SEND_FILE:
			setSendFile((String) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	/**
	 * @generated
	 */
	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case Platform_servicePackage.I_PLATFORM_SERVICE__DOWNLOAD:
			setDownload(null);
			return;
		case Platform_servicePackage.I_PLATFORM_SERVICE__UPLOAD_DATA:
			setUploadData(null);
			return;
		case Platform_servicePackage.I_PLATFORM_SERVICE__UPLOAD_FILE:
			setUploadFile(null);
			return;
		case Platform_servicePackage.I_PLATFORM_SERVICE__RECEIVE:
			setReceive(null);
			return;
		case Platform_servicePackage.I_PLATFORM_SERVICE__SEND_COMMAND:
			setSendCommand(null);
			return;
		case Platform_servicePackage.I_PLATFORM_SERVICE__SEND_FILE:
			setSendFile(null);
			return;
		}
		super.eUnset(featureID);
	}

	/**
	 * @generated
	 */
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case Platform_servicePackage.I_PLATFORM_SERVICE__DOWNLOAD:
			return download != null;
		case Platform_servicePackage.I_PLATFORM_SERVICE__UPLOAD_DATA:
			return uploadData != null;
		case Platform_servicePackage.I_PLATFORM_SERVICE__UPLOAD_FILE:
			return uploadFile != null;
		case Platform_servicePackage.I_PLATFORM_SERVICE__RECEIVE:
			return receive != null;
		case Platform_servicePackage.I_PLATFORM_SERVICE__SEND_COMMAND:
			return sendCommand != null;
		case Platform_servicePackage.I_PLATFORM_SERVICE__SEND_FILE:
			return sendFile != null;
		}
		return super.eIsSet(featureID);
	}
} //Platform_Service
